src/engine/ItemmarkManager.ts

```diff
@@ -0,0 +1,105 @@
+import type { IMediaContainer } from './providers/MediaPlugin';
+import { type StorageController, Store } from './StorageController';
+
+export class ItemmarkManager {
+
+    private readonly Items:Map<string,ItemMark[]>=new Map();
+    constructor(private readonly storage: StorageController) {
+        this.Load();
+    }
+
+    private async Load() {
+        const mediamarks = await this.storage.LoadPersistent<ItemmarkSerialized[]>(Store.Itemmarks);
+        mediamarks.forEach(itemmark => {
+            this.Items.set(`${itemmark.Media.ProviderID} :: ${itemmark.Media.EntryID}`,itemmark.Marks);
+        });
+    }
+
+    private Serialize(media: IMediaContainer, marks: ItemMark[]): ItemmarkSerialized {
+        return {
+            Media: {
+                ProviderID: media.Parent.Parent.Identifier,
+                EntryID: media.Parent.Identifier
+            },
+            Marks: marks,
+        };
+    }
+
+    /* Retrieves the marks of a media */
+    public Find(entry: IMediaContainer): ItemMark[] {
+        return this.Items.get(this.StorageKey(entry));
+    }
+
+    /* Add a mark of an item */
+    public async Add(entry: IMediaContainer, kind: MarkType): Promise<ItemMark[]> {
+        const newmark: ItemMark = {
+            IdentifierHash: this.Hash(entry.Identifier),
+            TitleHash: this.Hash(entry.Title),
+            kind: kind,
+        };
+        const marks = this.Find(entry.Parent) ?? [];
+        marks.push(newmark);
+        const key = this.StorageKey(entry.Parent);
+        this.Items.set(key,marks);
+        await this.storage.SavePersistent<ItemmarkSerialized>(this.Serialize(entry, marks), Store.Itemmarks, key);
+        return marks;
+    }
+
+    /* Remove the mark of an item */
+    public async Remove(entryToRemove: IMediaContainer): Promise<ItemMark[]> {
```

Comment:
Caller probably has the mark at hand that shall be removed, return a list is not expected
`Remove(mark: ItemMark): Promise<bool>`

Reply:
this allows the UI that is using the list to not refilter the marks by simply reassigning the results already filtered

Reply:
I see the intention, but business logic shoud be genral purpose, just ignore the fact that it will b consumed by a frontend.
May introduce antoher method to acquire the updated list of marks.